src/test/java/io/vertx/core/http/HttpBandwidthLimitingTest.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2011-2021 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+package io.vertx.core.http;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import io.vertx.core.net.TrafficShapingOptions;
+import org.apache.commons.lang.time.StopWatch;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.test.core.TestUtils;
+
+@RunWith(Parameterized.class)
+public class HttpBandwidthLimitingTest extends Http2TestBase {
+  private static final int OUTBOUND_LIMIT = 64 * 1024;  // 64KB/s
+  private static final int INBOUND_LIMIT = 64 * 1024;   // 64KB/s
+  private static final int TEST_CONTENT_SIZE = 64 * 1024 * 4;   // 64 * 4 = 256KB
+
+  private final File sampleF = new File("src/test/resources/test_traffic.txt");
```

Comment:
instead use the constant `TestUtils.MAVEN_TARGET_DIR` as base directory and then the resource should be in `classes/test_traffic.txt` , you should replace `/` by the file separator so test can run on windows